pkgs/development/tools/ilspy/default.nix

```diff
@@ -0,0 +1,45 @@
+{ pkgs ? (import <nixpkgs> {})}:
+with pkgs;
+let 
+  libs = [
+    xorg.libX11
+    stdenv.cc.cc.lib
+    kerberos
+    curl
+    lttng-ust
+    gtk3
+    openssl.out
+    icu
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "ilspy";
+  version = "5.0-rc2";
+
+  nativeBuildInputs = [ 
+    autoPatchelfHook
+  ];
+
+  dontStrip = true;
+  sourceRoot = ".";
+
+  buildInputs = [
+    makeWrapper
+    unzip
+  ] ++ libs;
+
+  src = fetchurl {
+    url = "https://github.com/icsharpcode/AvaloniaILSpy/releases/download/v${version}/ILSpy-linux-x64-Release.zip";
+    sha256 = "1p1fkdkpbcqj2q4n7y241lzx98w33vmh4n2gahw7lbrrg1p7rxxa";
+  };
+
+  installPhase = ''
+    chmod a+x ILSpy
+    mkdir -p $out/bin
+    mv * $out/bin
+
+    wrapProgram $out/bin/ILSpy \
+      --prefix LD_LIBRARY_PATH : "${stdenv.lib.makeLibraryPath libs}"
```

Comment:
Added one in https://github.com/NixOS/nixpkgs/pull/91255/commits/e85a1c107dc636a4f508ad13ac78f20121fbd13d but tbh I have no idea what I'm doing here... I use i3 so it's been many years since I even consumed these things I think.